tube/etl/indexers/base/parser.py

```diff
@@ -91,7 +91,8 @@ def get_es_types(self):
                 array_type_condition = is_array_type or has_array_agg_fn
                 if array_type_condition:
                     types[p.name] = (self.select_widest_type(p.type), 1)
-                    self.array_types.append(p.name)
+                    if p.name not in self.array_types:
```

Comment:
This comment is probably not directly linked to this PR, but I was just thinking based on the use cases of `array_types`. Almost all the time we either verify whether it is in the array or not(except [here](https://github.com/uc-cdis/tube/blob/d3ee74545f2ad8b8fb91700e2bbc3c5d2789c012/tube/etl/outputs/es/writer.py#L102-L102) ). So if we are not particular about the order of the items in the list, can we replace this to a `set`?

By doing that we don't have to worry about duplicates and also increase the search performance to O(1).

Reply:
array_type can be either list of set. Since there are cases where we want the duplicates. The `in` operation is optimized by the data type